src/transformers/models/maskformer/feature_extraction_maskformer.py

```diff
@@ -0,0 +1,549 @@
+              model (when `annotations` are provided)
+            - **class_labels** -- Optional class labels of shape `(batch_size, num_classes) to be fed to a model (when
+              `annotations` are provided)
+        """
+
+        max_size = self._max_by_axis([list(image.shape) for image in pixel_values_list])
+        c, height, width = max_size
+        pixel_values = []
+        pixel_mask = []
+        mask_labels = []
+        class_labels = []
+
+        for idx, image in enumerate(pixel_values_list):
+            # create padded image
+            if pad_and_return_pixel_mask:
+                padded_image = np.zeros((c, height, width), dtype=np.float32)
+                padded_image[: image.shape[0], : image.shape[1], : image.shape[2]] = np.copy(image)
+                image = padded_image
+            pixel_values.append(image)
+            # if we have a target, pad it
+            if annotations:
+                annotation = annotations[idx]
+                masks = annotation["masks"]
+                if pad_and_return_pixel_mask:
+                    padded_masks = np.zeros((masks.shape[0], height, width), dtype=masks.dtype)
+                    padded_masks[:, : padded_masks.shape[1], : padded_masks.shape[2]] = np.copy(padded_masks)
+                    masks = padded_masks
+                mask_labels.append(masks)
+                class_labels.append(annotation["labels"])
+            if pad_and_return_pixel_mask:
+                # create pixel mask
+                mask = np.zeros((height, width), dtype=np.int64)
+                mask[: image.shape[1], : image.shape[2]] = True
+                pixel_mask.append(mask)
+
+        # return as BatchFeature
+        data = {
+            "pixel_values": pixel_values,
+            "pixel_mask": pixel_mask,
+        }
+
+        if annotations:
+            data["mask_labels"] = mask_labels
+            data["class_labels"] = class_labels
+
+        encoded_inputs = BatchFeature(data=data, tensor_type=return_tensors)
+
+        return encoded_inputs
+
+    def post_process_segmentation(
+        self, outputs: MaskFormerForInstanceSegmentationOutput, target_size: Tuple[int, int] = None
+    ) -> Tensor:
+        """Converts the output of [`MaskFormerForInstanceSegmentationOutput`] into image segmentation predictions. Only supports
+        PyTorch.
+
+                Args:
+                    outputs (MaskFormerForInstanceSegmentationOutput): The outputs from MaskFor
+
+                Returns:
+                    Tensor: A tensor of shape `batch_size, num_labels, height, width`
+        """
+        # class_queries_logitss has shape [BATCH, QUERIES, CLASSES + 1]
+        class_queries_logits = outputs.class_queries_logits
+        # masks_queries_logits has shape [BATCH, QUERIES, HEIGHT, WIDTH]
+        masks_queries_logits = outputs.masks_queries_logits
+        if target_size is not None:
+            masks_queries_logits: Tensor = interpolate(
```

Comment:
We can't have `Tensor` annotations everywhere because `Tensor` should only be imported when torch is available. We also discussed to keep type annotations for complex types only.